### PR TITLE
Updates to adapt to newer Dawn with Metal

### PR DESCRIPTION
**Pull request by dneto0**
- Dawn device is always taken from the engine config
- Fix buffer view copy height
- Clear color is stored as a ::dawn::Color value rather than ClearColorCommand

This now builds and links against fresh Dawn with Metal.  But  the clear.amber example fails with Verifier::Probe, it says texel's format is nullptr.

### CODE REVIEW

**Comment by dneto0** [comment]
Now this passes the clear test again on Dawn over Metal.

File: samples/config_helper_dawn.cc [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    uint32_t,
+    const std::vector<std::string>&,
+    const std::vector<std::string>&,
+    bool,
+    std::unique_ptr<amber::EngineConfig>* config) {
+#if AMBER_DAWN_METAL
+  auto r = dawn::CreateMetalDevice(&dawn_instance_, &dawn_device_);
+  if (!r.IsSuccess())
+    return r;
+#else
```
Comment (jaebaek) [inline samples/config_helper_dawn.cc]: Does this mean Amber supports Dawn **only on Metal** after merging this CL?

Comment (dneto0) [reply samples/config_helper_dawn.cc]: Yes.
But currently Amber doesn't support anything on Dawn, by my reading of the code.
I've only tested this new PR on Dawn on Metal.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -106,6 +106,10 @@ if (${AMBER_ENABLE_TESTS})
       -Wno-weak-vtables
     )
   endif()
+  if (${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
+    # EXPECT_EQ(0u, ...) triggers a "zero as null pointer constant" warning.
```
[the remark this answers is not in the document]
Comment (dneto0) [reply src/CMakeLists.txt]: I deleted this.  I didn't need it this time around ?!

File: samples/dawn_device_metal.mm [hunk 3]
```diff
@@ -0,0 +1,49 @@
+// Copyright 2018 The Amber Authors.
```
[the remark this answers is not in the document]
Comment (dneto0) [reply samples/dawn_device_metal.mm]: Fixed.

File: samples/CMakeLists.txt [hunk 4]
```diff
@@ -26,10 +26,24 @@ if (${Vulkan_FOUND})
...
+  else()
+    add_definitions(-DAMBER_DAWN_METAL=0)
+  endif()
+endif()
+
 add_executable(amber ${AMBER_SOURCES})
 target_include_directories(amber PRIVATE "${CMAKE_CURRENT_BINARY_DIR}/..")
 set_target_properties(amber PROPERTIES OUTPUT_NAME "amber")
-target_link_libraries(amber libamber ${VULKAN_LIB})
+target_link_libraries(amber libamber ${VULKAN_LIB} ${AMBER_EXTRA_LIBS})
```
[the remark this answers is not in the document]
Comment (dneto0) [reply samples/CMakeLists.txt]: Done

**Pull request merged**